app/models/episode.rb

```diff
@@ -101,6 +101,17 @@ def image
     images.complete.first
   end
 
+  # API updates for image=
+  def image_file; images.first; end
+  def image_file=(file)
+    img = EpisodeImage.build(file)
+    if img && img.original_url != image_file.try(:original_url)
+      images << img
+    elsif !img
+      images.destroy_all
```

Comment:
You'll see similar methods for itunes and feed images.  These do similar logic as episode `media_files`, and the same logic the CMS handlers did:

- Check if we already have an image with the same original url - if so, skip
- If not, create a new image but leave the old one there.  The new image only replaces the old after the Porter callbacks complete.
- If null, immediately delete all images.